[Lens] Edit data view in flyout

flash1293 opened the pull request:
Part of https://github.com/elastic/kibana/issues/141806

If the app supports it, "Manage this data view" will open the flyout instead of navigating away:
<img width="1280" alt="Screenshot 2022-09-30 at 16 16 30" src="https://user-images.githubusercontent.com/1508364/193289784-87fe7b12-cf04-4d9f-9f23-33e14b476b70.png">

In this case there is a link below the heading to go to the management page if the user wants to do that.

Discover is currently changing the state handling in this area, that's why it's only implemented in Lens for now.

elasticmachine commented:
Pinging @elastic/kibana-vis-editors @elastic/kibana-vis-editors-external (Team:VisEditors)

stratoula commented:
@flash1293 this works fine. I have 2 questions:

- I open the flyout for an existing dataview and I change the pattern. The Save button is unresponsive. Shouldn't it update the current dataview? If not, maybe we should disable the button?
<img width="1491" alt="image" src="https://user-images.githubusercontent.com/17003240/193550883-8007c465-f107-4b23-a6b8-0430ee472c7f.png">

- When I create an adhoc dataview and then open the Edit flyout I see only the Save button. I expect that clicking it will save my dataview as permanent but it doesn't. When I refresh the page, the dataview is lost. Maybe this is just an update CTA? Should we change the text from Save to Update maybe in order to not be so confusing? Wdyt?
<img width="738" alt="image" src="https://user-images.githubusercontent.com/17003240/193553273-3c9e49ff-96bd-4ee2-9b23-06dcf83fd0cc.png">

flash1293 commented:
Thanks for your review @stratoula !

> I open the flyout for an existing dataview and I change the pattern. The Save button is unresponsive. Shouldn't it update the current dataview? If not, maybe we should disable the button?

That was a bug in the data view editor plugin, I fixed it

> When I create an adhoc dataview and then open the Edit flyout I see only the Save button. I expect that clicking it will save my dataview as permanent but it doesn't. When I refresh the page, the dataview is lost. Maybe this is just an update CTA

Good point - I changed the button label to "Continue to use without saving". It was meant to keep it as an ad-hoc data view but update it in-place.

flash1293 commented:
I also pulled in https://github.com/elastic/kibana/pull/142562/files to fix another problem I noticed with not all parts of the UI updating correctly.

Dosant commented on src/plugins/data_view_editor/public/components/data_view_editor_flyout_content.tsx:
```diff
@@ -376,6 +385,17 @@ const IndexPatternEditorFlyoutContentComponent = ({
         <EuiTitle data-test-subj="flyoutTitle">
           <h2>{editData ? editorTitleEditMode : editorTitle}</h2>
         </EuiTitle>
+        {showManagementLink && editData && editData.id && (
```
I think we should hide it when you edit an ad-hoc data view (now we show it when you edit ad-hoc data view)

flash1293 replied:
Good point, added that

Dosant reviewed (approved):
Code lgtm, 

noticed some minor things during testing
